### ISSUE TITLE
Importing large export assignment object causes Maximum call stack size exceeded

### ISSUE THREAD

**Issue by AnyhowStep** [microsoft/TypeScript#31824]
**TypeScript Version:**  3.5.1

**Search Terms:** import, export assignment, `export =`, call stack

**Code**

This crashes `tsc`,

```ts
//npm install --save type-mapping@1.1.2
//`ts-node ./index.ts` works fine
//`tsc` gives `Maximum call stack size exceeded`
//`"type-mapping/fluent"` is a file that uses `export =`
//to export an object with many properties
import * as tm from "type-mapping/fluent";
console.log(tm);
```

This works fine,

```ts
//npm install --save type-mapping@1.1.2
//`ts-node ./index.ts` works fine
//`tsc` also works fine
//`"type-mapping"` is a file that uses `export *`
//to export many objects, even more than "type-mapping/fluent"
import * as tm from "type-mapping";
console.log(tm);
```

**Expected behavior:**

`tsc` should compile fine in both cases.

`"type-mapping/fluent"` uses `export =` to export a single object with many properties (141 properties).

However, the number of "things" exported is less than the amount exported
by `"type-mapping"`, which uses `export *` (178, excluding interfaces and typedefs).

So, this makes me suspect that the problem is `export =`.
However, I simply have to use `export =` because `"type-mapping/fluent"` is generating these properties during run-time. (During compile-time, the export is a mapped object type)

**Actual behavior:**

`Maximum call stack size exceeded` with `"type-mapping/fluent"` and `export =`.

Works fine with `"type-mapping"` and `export *`

**Related Issues:** Not that I could find

**Comment by AnyhowStep** [comment]
The export is created here,
https://github.com/AnyhowStep/type-mapping/blob/1a999792f00fe25d2d3d1038d48b857a82b94219/src/fluent-lib/non-generic.ts

It is exported here,
https://github.com/AnyhowStep/type-mapping/blob/1a999792f00fe25d2d3d1038d48b857a82b94219/src/fluent-lib/index.ts

And here,
https://github.com/AnyhowStep/type-mapping/blob/1a999792f00fe25d2d3d1038d48b857a82b94219/fluent.d.ts

**Comment by AnyhowStep** [comment]
I just made a workaround for large `export =` objects causing `tsc` to crash.

I wrote a script to generate a file that uses `export *`, instead of `export =`

https://github.com/AnyhowStep/type-mapping/blob/13d9a96241d91d0e486ee4e8194ad163cce3986c/script/src/generate-fluent-lib-export.ts

I just tested it and it seems to work without crashing now.
I'll have to test some more and see if there are any lingering side-effects.
